jax/_src/test_util.py

```diff
@@ -1211,12 +1211,29 @@ def assertMultiLineStrippedEqual(self, expected, what):
     self.assertMultiLineEqual(expected_clean, what_clean,
                               msg=f"Found\n{what}\nExpecting\n{expected}")
 
+
   @contextmanager
   def assertNoWarnings(self):
-    with warnings.catch_warnings():
-      warnings.simplefilter("error")
+    with test_warning_util.raise_on_warnings():
       yield
 
+  @contextmanager
+  def assertWarnsRegex(self, expected_warning, expected_regex):
```

Comment:
The existing `assertWarnsRegex` can be used either as a context manager or as a simple function call, passing the callable and args/kwargs directly. Should we support that in the new version?